ci/start_init_test.sh

```diff
@@ -837,10 +837,31 @@ run_data_integrity_test() {
 	#Cleanup is not being performed here because this data will be used to test clone feature in the next test
 }
 
-create_snapshot() {
-	echo "--------------create_snapshot-------------"
-	id=`curl http://$1:9501/v1/volumes | jq '.data[0].id' |  tr -d '"'`
-	curl -H "Content-Type: application/json" -X POST -d '{"name":"snap1"}' http://$CONTROLLER_IP:9501/v1/volumes/$id?action=snapshot
+# create_and_verify_snapshot first creates a snapshot and verifies if it
+# is of type snapshotOutput in case of failure it returns type:error which
+# is success case that the snapshot snap1 does not exists.
+# As a part of negative test we are trying to create the snapshot with the
+# same name which returns snapshot_exists.
+create_and_verify_snapshot() {
+	echo "--------------create_and_verify_snapshot-------------"
+       snapshot_exists="Snapshot: snap1 already exists"
+       id=`curl http://$CONTROLLER_IP:9501/v1/volumes | jq '.data[0].id' |  tr -d '"'`
+       type=`curl -H "Content-Type: application/json" -X POST -d '{"name":"snap1"}' http://$CONTROLLER_IP:9501/v1/volumes/$id?action=snapshot | jq '.type' | tr -d '"'`
+       if [ "$type" != "snapshotOutput" ];
+       then
+              echo "create and verify snapshot test1 failed"
+              collect_logs_and_exit
+       else
+              echo "create and verify snapshot test1 passed"
+       fi;
+       message=`curl -H "Content-Type: application/json" -X POST -d '{"name":"snap1"}' http://$CONTROLLER_IP:9501/v1/volumes/$id?action=snapshot | jq '      .message' | tr -d '"'`
+       if [ "$message" == "$snapshot_exists" ];
+       then
+              echo "create and verify snapshot test2 passed"
+       else
+              echo "create and verify snapshot test2 failed"
+              collect_logs_and_exit
+       fi;
 }
```

Comment:
Let's create one more snapshot with a different name and check if something is breaking.
Lets also create, delete and recreate a snapshot with the same name and test.

Reply:
Thanks for the review. I have updated the PR with the above requested changes.